src/wp-includes/comment-template.php

```diff
@@ -1927,16 +1927,24 @@ function post_reply_link( $args = array(), $post = null ) {
  *
  * @since 2.7.0
  *
- * @param string $text Optional. Text to display for cancel reply link. If empty,
- *                     defaults to 'Click here to cancel reply'. Default empty.
+ * @param string $text    Optional. Text to display for cancel reply link. If empty,
+ *                        defaults to 'Click here to cancel reply'. Default empty.
+ * @param int    $post_id Optional. The post ID the comment thread is being
+ *                        displayed for. Default current global post ID.
  * @return string
  */
-function get_cancel_comment_reply_link( $text = '' ) {
+function get_cancel_comment_reply_link( $text = '', $post_id = 0 ) {
 	if ( empty( $text ) ) {
 		$text = __( 'Click here to cancel reply.' );
 	}
 
-	$style = isset( $_GET['replytocom'] ) ? '' : ' style="display:none;"';
+	if ( 0 === $post_id ) {
+		$post_id = get_the_ID();
+	}
+
+	$reply_to_id = _get_comment_reply_id( $post_id );
```

Comment:
```suggestion
	$post = get_post( $post );
	if ( ! $post ) {
		return '';
	}
	
	$post_id = $post->ID;
	$reply_to_id = _get_comment_reply_id( $post_id );
```

The goal is to bail if the post ID is invalid. If you think it needs the undisplayed link markup, that's cool too.